### ISSUE TITLE
The C header file cannot be generated correctly.

### ISSUE THREAD

**Issue by ssrlive** [mozilla/cbindgen#935]
The c header file cannot be generated correctly. [This](https://github.com/ssrlive/socks-hub) is my project.
I entered the project root directory and entered the command `cbindgen -l c`, and got the following output.
I did not see the expected function prototype `socks_hub_run` and so on.
```
#include <stdarg.h>
#include <stdbool.h>
#include <stdint.h>
#include <stdlib.h>
```

**Comment by ssrlive** [comment]
emm, I have found the reason, it's about `cfg_if::cfg_if!` macro. when I remove it, cbindgen works fine.

can you fix it as a bug? 


![image](https://github.com/mozilla/cbindgen/assets/30760636/d949f2c0-069a-4032-b5c0-6f5aa4539b32)

**Comment by emilio** [comment]
There's the cargo expand feature that should work, right?

**Comment by luodongseu** [comment]
mark

**Comment by emilio** [comment]
Yes, so to clarify on my previous comment. It is expected that cbindgen doesn't understand macros (and that includes cfg_if), unless you use the cargo expand feature.

**Comment by ssrlive** [comment]
I don't know how to use cargo expand feature.

**Comment by emilio** [comment]
See `[parse.expand]` [here](https://github.com/mozilla/cbindgen/blob/master/docs.md).

**Comment by ssrlive** [comment]
No work.

![image](https://github.com/user-attachments/assets/e43ea7a2-7724-471f-8437-9538d496d1eb)
